csrc/fusion_profiler.cpp

```diff
@@ -0,0 +1,737 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <cxxabi.h>
+#include <fusion_profiler.h>
+#include <iomanip>
+
+namespace nvfuser {
+
+namespace {
+
+// Copying some code from the CUPTI samples/common code
+// CUPTI buffer size 4.0 KB
+#define BUF_SIZE (1 * 4 * 1024)
+// 8-byte alignment for the buffers
+#define ALIGN_SIZE (8)
+#define ALIGN_BUFFER(buffer, align)                                 \
+  (((uintptr_t)(buffer) & ((align)-1))                              \
+       ? ((buffer) + (align) - ((uintptr_t)(buffer) & ((align)-1))) \
+       : (buffer))
+
+const char* get_demangled_name(const char* pName) {
```

Comment:
we also have `demangle` in `csrc/exceptions.cpp`

Should we just put them together. Also __cxa_demangle may fail. So either assert or just return the original there.

Reply:
https://github.com/NVIDIA/Fuser/blob/ae428309dc41c987c9ed7673e8e832a5828ad1a4/csrc/exceptions.cpp#L51

Reply:
I moved to use `demange()` from `csrc/exceptions.cpp`.  Therefore, I had to move the code out of the anonymous `namespace`.